src/AzureRMTools.ts

```diff
@@ -165,6 +167,33 @@ export class AzureRMTools {
                 source = source ?? vscode.window.activeTextEditor?.document.uri;
                 await openTemplateFile(this._mapping, source, undefined);
             });
+        registerCommand("azurerm-vscode-tools.insertItem", async (_context: IActionContext, uri?: vscode.Uri, editor?: vscode.TextEditor) => {
+            editor = editor || vscode.window.activeTextEditor;
+            uri = uri || vscode.window.activeTextEditor?.document.uri;
+            // If "Sort template..." was called from the context menu for ARM template outline
+            if (typeof uri === "string") {
+                uri = vscode.window.activeTextEditor?.document.uri;
+            }
+            if (uri && editor) {
+                const sortType = await ext.ui.showQuickPick(getInsertItemType(), { placeHolder: 'What do you want to insert?' });
+                await this.insertItem(sortType.value, uri, editor);
+            }
+        });
+        registerCommand("azurerm-vscode-tools.insertParameter", async () => {
+            await this.insertItem(SortType.Parameters);
```

Comment:
Rename SortType enum to  something like TemplateSectionType

Reply:
Fixed.